### ISSUE TITLE
Fix JSON support

### ISSUE THREAD

**Issue by TheBoneJarmer** [TheBoneJarmer/TiledCS#22]
Up till now the idea was to have the tiled maps being (de)serialized from JSON string input.. However, it has become clear to me  that this approach may have been enough for earlier versions but I started realizing more and more differences between the XML format and the JSON format. Hence, simply (de)serializing won't cut it anymore. Therefore I would like to use a different approach.

With NewtonSoft.Json there is also the possibility to parse a JSON file just the way I am doing now with XML. This does require however that I have to rewrite both the TiledMap and TiledTileset classes entirely as parsing now have to happen in two directions.

**Comment by vpenades** [comment]
If you ever switch to json, don't use newtonsoft, use System.Text.Json which is the new kid on the block, and future proof json serialization library.

**Comment by TheBoneJarmer** [comment]
Hi @vpenades 

Thanks for the recommendation! Just out of curiosity though, why should I prefer System.Text.Json over NewtonSoft? Because the former has proven itself stable and has been the standard for json parsing for like a decade. Heck, in my career I never even considered using something else since it simply works.

With kind regards,
TheBoneJarmer

**Comment by vpenades** [comment]
System.Text.Json has mostly the same funcionality of Newtonsoft, so in that aspect there's not much difference.

But, on large projects, it is fairly common to depend on libraries that also depend on _different versions_ of newtonsoft, which is a source of conflicts.

System.Text.Json addresses this issue, because in Net5 and above, it is implicitly included (you don't need to reference any nuget at all)

Long story short: the less dependencies, the better.

**Comment by TheBoneJarmer** [comment]
That actually sounds like a very legit reason. Thanks! It is true that with more dependencies, which also use NewtonSoft.Json, more conflicts rise. That said, I will have to further investigate this since I will have to do some more research on backwards compatibility because you mentioned .NET 5. So the first question that pops to my mind is if this feature is also supported on older versions of .NET.

Sadly, not everyone has the luxury or opportunity to simply switch to the newer .NET SDK. So if this feature exists only from .NET 5, I will have to pass. Otherwise it is worth looking into since it does mean that TiledCS can be truly stand-alone without any need of 3rd party dependencies.

**Comment by vpenades** [comment]
For older version, I usually do multi targeting:

```xml
<PropertyGroup>    
    <TargetFrameworks>netstandard2.0;netstandard2.1;netcoreapp3.1;net5.0</TargetFrameworks>
  </PropertyGroup>
  
  <ItemGroup Condition=" '$(TargetFramework)' != 'net5.0' ">    
    <PackageReference Include="System.Text.Json" Version="5.0.2" />
  </ItemGroup>
```

**Comment by TheBoneJarmer** [comment]
Oh sweet, thanks. Did not realize System.Text.Json was a NuGet package too. So if I understand correctly, from .NET 5 and onward, this functionality if part of the .NET SDK. But for older versions it is not? So for older versions I need to reference the NuGet package instead. Ok, that sounds like a plan to me.

**Comment by vpenades** [comment]
Yes, same story as with System.Numerics.Vectors, System.Buffers, and so on

**Comment by TheBoneJarmer** [comment]
I am going to close this one. It is a lot of work to implement this manually, let alone being easy. So far nobody requested this actually and everyone uses the xml format anyway as far as I have noticed.
